### PR TITLE
chore: (enzyme replacement in test files) : User Story 2130867

### PR DESCRIPTION
**Pull request by rohittarpara**
Details
Enzyme replacement done for total 14 files.

Motivation
enzyme replacement in files under src/tests/unit/tests/common/component/card

Associated User story - [USER STORY 2130867](https://dev.azure.com/mseng/1ES/_workitems/edit/2130867)

Context
All changes are done in test files only and snapshots are updated.

#### Pull request checklist

- [x] Addresses an existing issue:  [USER STORY 2130867](https://dev.azure.com/mseng/1ES/_workitems/edit/2130867)
- [x] Ran `yarn fastpass`
- [x] Added/updated relevant unit test(s) (and ran `yarn test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] PR title *AND* final merge commit title both start with a semantic tag (`fix:`, `chore:`, `feat(feature-name):`, `refactor:`). See `CONTRIBUTING.md`.
- [N/A] (UI changes only) Added screenshots/GIFs to description above
- [N/A] (UI changes only) Verified usability with NVDA/JAWS

### CODE REVIEW

File: src/tests/unit/tests/common/components/cards/instance-details.test.tsx [hunk 1]
```diff
@@ -66,62 +73,70 @@ describe('InstanceDetails', () => {
...
         cardSelectionMessageCreatorMock
-            .setup(mock => mock.toggleCardSelection(It.isAnyString(), It.isAnyString(), eventStub))
+            .setup(mock => mock.toggleCardSelection(It.isAnyString(), It.isAnyString(), It.isAny()))
             .verifiable(Times.once());
 
-        const wrapper = shallow(<InstanceDetails {...props} />);
-        const element = wrapper.find(`.${styles.hiddenHighlightButton}`);
-        expect(element.length).toBe(1);
+        const renderResult = render(<InstanceDetails {...props} />);
+        const element = renderResult.container.querySelector(`.${styles.hiddenHighlightButton}`);
```
Comment (madalynrose) [inline src/tests/unit/tests/common/components/cards/instance-details.test.tsx:111]:
```suggestion
        const element = renderResult.getByRole('button');
```

File: src/tests/unit/tests/common/components/cards/instance-details.test.tsx [hunk 2]
```diff
@@ -66,62 +73,70 @@ describe('InstanceDetails', () => {
...
     it('applies focused styling on card when hidden highlight button is focused', () => {
         (React.useRef as jest.Mock).mockReturnValue(hiddenButtonRefStub);
         setupGetPropertyConfigByIdMock();
-        const wrapper = shallow(<InstanceDetails {...props} />);
-        const button = wrapper.find(`.${styles.hiddenHighlightButton}`);
-        expect(button.length).toBe(1);
+        const renderResult = render(<InstanceDetails {...props} />);
+        const button = renderResult.container.getElementsByClassName(
+            `${styles.hiddenHighlightButton}`,
+        );
```
Comment (madalynrose) [inline src/tests/unit/tests/common/components/cards/instance-details.test.tsx:129]:
```suggestion
        const button = renderResult.getAllByRole('button');
```

**Review by madalynrose — CHANGES_REQUESTED** [review]
A couple small suggestions but looks great!

File: src/tests/unit/tests/common/components/cards/__snapshots__/cards-visualization-modifier-buttons.test.tsx.snap [hunk 3]
```diff
@@ -1,30 +1,104 @@
...
-    allCardsCollapsed={true}
-    cardSelectionMessageCreator={{}}
-    visualHelperEnabled={true}
-  />
-</div>
+<DocumentFragment>
+  <div
+    class="cardsVisualizationModifiersContainer"
+  >
+    <button
```
Comment (madalynrose) [inline src/tests/unit/tests/common/components/cards/__snapshots__/cards-visualization-modifier-buttons.test.tsx.snap:8]: I would recommend mocking the `ExpandCollapseAllButton` component for this test so there is less irrelevant HTML in the snapshot.

File: src/tests/unit/tests/common/components/cards/__snapshots__/instance-details.test.tsx.snap [hunk 4]
```diff
@@ -1,305 +1,123 @@
...
+        <button
+          aria-label="highlight test-id"
+          aria-pressed="true"
+          class="hiddenHighlightButton"
+        />
+        <mock-instancedetailsfooter
+          deps="[object Object]"
+          result="[object Object]"
+        />
+      </div>
```
Comment (madalynrose) [inline src/tests/unit/tests/common/components/cards/__snapshots__/instance-details.test.tsx.snap:28]: We should make sure we use the `expectMockedComponentPropsToMatchSnapshots` helper so we don't miss this information.

Comment (rohittarpara) [reply src/tests/unit/tests/common/components/cards/__snapshots__/instance-details.test.tsx.snap:196]: On every InstanceDetails component render, after that i set below code, please review and let me know if you have any feedback.


const renderResult = render(<InstanceDetails {...props} />);
expectMockedComponentPropsToMatchSnapshots([InstanceDetailsFooter]);

File: src/tests/unit/tests/common/components/cards/__snapshots__/cards-visualization-modifier-buttons.test.tsx.snap (hunk 3 above)
Comment (rohittarpara) [reply src/tests/unit/tests/common/components/cards/__snapshots__/cards-visualization-modifier-buttons.test.tsx.snap]: Done, please review on the latest commit.

File: src/tests/unit/tests/common/components/cards/instance-details.test.tsx (hunk 2 above)
Comment (rohittarpara) [reply src/tests/unit/tests/common/components/cards/instance-details.test.tsx]: Done, please review on the latest commit.

File: src/tests/unit/tests/common/components/cards/instance-details.test.tsx (hunk 1 above)
Comment (rohittarpara) [reply src/tests/unit/tests/common/components/cards/instance-details.test.tsx]: Done, please review on the latest commit.

**Review by madalynrose — APPROVED** [review]

**Pull request merged**